### DISCUSSION TITLE
Using modules and environments together

### DISCUSSION THREAD

**Discussion by keith-miller** [gruntwork-io/knowledge-base#850] [Help]
Hello all! I'm new to Terragrunt (coming from Terraform) and trying to set up my first project. I have the following directory structure:

![image](https://github.com/user-attachments/assets/3904cd7b-344b-4e10-8242-a2c780ad2262)

And in `environments/dev/terragrunt.hcl` I have:

```hcl
include {
  path = find_in_parent_folders()
}

terraform {
  source = "../../modules//mosaic"
}

```

But when I run:

```shell
environments\dev > terragrunt run-all plan
```

I get: `No changes. Your infrastructure matches the configuration.`

I'm just wondering how I correct use modules and environments together. Thanks!

**Comment by keith-miller** [comment]
I got it working by making matching `terragrunt.hcl` files for the modules I want to use under `environments/dev` then running the terragrunt commands from there:

![image](https://github.com/user-attachments/assets/0ca77887-75ce-483b-a4d2-06b776a59825)

I'll have to copy those for each environment so if there is a better way please let me know!

**Reply by yhakbar** [reply]
Hey @keith-miller , did you get a chance to go through the [Getting Started Guide](https://terragrunt.gruntwork.io/docs/#getting-started)?
